pkg/api/controller.go

```diff
@@ -817,12 +841,27 @@ func (c *Controller) CreateUser(w http.ResponseWriter, r *http.Request, body Cre
 		FriendlyName: nil,
 		Source:       "internal",
 	}
+	normalizedEmail := normalizeAndSetToLowercase(body.Id)
+	invite := swag.BoolValue(body.InviteUser)
+
+	if invite {
+		setUsernameAndEmail(u, normalizedEmail)
+	}
 	ctx := r.Context()
 	c.LogAction(ctx, "create_user")
 	_, err := c.Auth.CreateUser(ctx, u)
+
 	if handleAPIError(w, err) {
 		return
 	}
+
+	if invite {
+		err = c.inviteUserRequest(ctx, normalizedEmail)
+		if handleAPIError(w, err) {
```

Comment:
Saw your answer about changing the api - which I don't think it is a big change.
Keep the current implementation the client will have to delete a user which it doesn't know the user ID after in the case we fail to send an email.

Reply:
Got it, will be in a separate PR, by how will be decided.